src/components/v1/V1Project/ProjectActivity/index.tsx

```diff
@@ -1,307 +1,61 @@
 import { DownloadOutlined } from '@ant-design/icons'
 import { t, Trans } from '@lingui/macro'
 import { Button, Space } from 'antd'
-import AddToBalanceEventElem from 'components/activityEventElems/AddToBalanceEventElem'
-import BurnEventElem from 'components/activityEventElems/BurnEventElem'
-import DeployedERC20EventElem from 'components/activityEventElems/DeployedERC20EventElem'
-import PayEventElem from 'components/activityEventElems/PayEventElem'
-import ProjectCreateEventElem from 'components/activityEventElems/ProjectCreateEventElem'
-import RedeemEventElem from 'components/activityEventElems/RedeemEventElem'
+import { AnyProjectEvent } from 'components/activityEventElems'
+import { JuiceListbox } from 'components/inputs/JuiceListbox'
 import Loading from 'components/Loading'
 import SectionHeader from 'components/SectionHeader'
 import { PV_V1 } from 'constants/pv'
 import { ProjectMetadataContext } from 'contexts/shared/ProjectMetadataContext'
 import { V1ProjectContext } from 'contexts/v1/Project/V1ProjectContext'
-import { useInfiniteSubgraphQuery } from 'hooks/useSubgraphQuery'
-import { SGWhereArg } from 'models/graph'
-import { PrintReservesEvent } from 'models/subgraph-entities/v1/print-reserves-event'
-import { TapEvent } from 'models/subgraph-entities/v1/tap-event'
-import { V1ConfigureEvent } from 'models/subgraph-entities/v1/v1-configure'
-import { AddToBalanceEvent } from 'models/subgraph-entities/vX/add-to-balance-event'
-import { DeployedERC20Event } from 'models/subgraph-entities/vX/deployed-erc20-event'
-import { PayEvent } from 'models/subgraph-entities/vX/pay-event'
-import { ProjectCreateEvent } from 'models/subgraph-entities/vX/project-create-event'
-import { ProjectEvent } from 'models/subgraph-entities/vX/project-event'
-import { RedeemEvent } from 'models/subgraph-entities/vX/redeem-event'
+import { ProjectEventFilter, useProjectEvents } from 'hooks/useProjectEvents'
 import { useContext, useMemo, useState } from 'react'
 
-import { JuiceListbox } from 'components/inputs/JuiceListbox'
-import ReservesEventElem from './eventElems/ReservesEventElem'
-import TapEventElem from './eventElems/TapEventElem'
-import V1ConfigureEventElem from './eventElems/V1ConfigureEventElem'
 import { V1DownloadActivityModal } from './V1DownloadActivityModal'
 
-type EventFilter =
-  | 'all'
-  | 'pay'
-  | 'burn'
-  | 'addToBalance'
-  | 'redeem'
-  | 'withdraw'
-  | 'printReserves'
-  | 'deployERC20'
-  | 'projectCreate'
-  | 'configure'
-
 const PAGE_SIZE = 10
 
 export function V1ProjectActivity() {
   const { projectId } = useContext(ProjectMetadataContext)
   const { tokenSymbol } = useContext(V1ProjectContext)
+  const [isFetchingMore, setIsFetchingMore] = useState<boolean>()
```

Comment:
custom addition to replicate `isFetchingNextPage` functionality of `useQuery`, referenced in PR description